File: python/leetcode/etc/array_partition_1.py
# https://leetcode.com/problems/array-partition-i/

class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        nums = sorted(nums)
        result = 0

        while nums != []:
            first = nums.pop()
            second = nums.pop()
            result += min(first, second)

        return result

File: go/leetcode/etc/arrayPartition1.go
// https://leetcode.com/problems/array-partition-i

func arrayPairSum(nums []int) int {
    sort.Ints(nums)
    length := len(nums)
    result := 0
    for i := 0; i < length; i+=2 {
        min := nums[i]
        if nums[i+1] < min {
           min = nums[i+1]
        }
        result += min
    }
    return result
}
